main.js

```diff
@@ -12,9 +12,42 @@ window.addEventListener('load', trigger.bind(null, 'load'));
 window.addEventListener('load', trigger.bind(null, 'DOMContentLoaded'));
 document.addEventListener('DOMContentLoaded', trigger.bind(null, 'DOMContentLoaded'));
 
+document.onreadystatechange = function () {
+	if (document.readyState === 'complete') {
+		trigger('DOMContentLoaded');
+		trigger('load');
+	}
+};
+
 if (document.readyState === 'complete') {
-	trigger('load');
-	trigger('DOMContentLoaded');
-} else if (document.readyState === 'interactive') {
```

Comment:
Sure. The problem is, that in IE (I think for cached pages, or page refresh), interactive is triggered even when the DOM is loaded only the head (document.body is still undefined), and it causes errors on all modules which listen on o.DOMContentLoaded (document.body is undefined). More than that, when would be correct to trigger DOMContentLoaded, it is not triggered, which means the modules does not init at all.

It is correct that if o-autoinit is called after the whole document loaded, but before window.onload, then it will wait until window.onload is triggered.

There would be a fine-tuned solution (but not sure how safe it is), to ignore "interactive" if document.attachEvent exists (i.e. the browser is an IE). What do you think?